### PR TITLE
[PM-1338] `AuthRequest` Event Logs

### PR DESCRIPTION
**Pull request by justindbaur**
## Type of change

```
- [ ] Bug fix
- [x] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Add event logs for `AdminApproval` AuthRequests and additional validation logic for creating/updating them. 

## Code changes

* **src/Api/Auth/Controllers/AuthRequestsController.cs:** To be able to create an AuthRequest of type `AdminApproval` you must be an authenticated user so that we can gather the current users organizations. I added protection on the non-authed endpoint and added a new authed endpoint to enable the creation of these requests.
* **src/Core/Auth/Services/Implementations/AuthRequestService.cs:**
* **src/Core/Auth/Settings/IPasswordlessAuthSettings.cs:** Added settings for expiration time customization.
* **src/Core/Enums/EventType.cs: ** New event types for organization actions regarding AuthRequests
* **test/Common/Helpers/BitAutoDataAttributeHelpers.cs:** Added the possibility of conversion of common types from string so you can define `TimeSpan`, `Guid`, etc in their string format. 
* **test/Core.Test/Auth/Services/AuthRequestServiceTests.cs:** Added loads of tests and wrote their story they are covering in plain english so that it can be a little easier to test.

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: test/Core.Test/Auth/Services/AuthRequestServiceTests.cs [hunk 1]
```diff
@@ -355,39 +583,96 @@ await sutProvider.GetDependency<IPushNotificationService>()
             async () => await sutProvider.Sut.UpdateAuthRequestAsync(authRequest.Id, authRequest.UserId, updateModel));
     }
 
+    /// <summary>
+    /// Story: Once the destiny of an AuthRequest has been decided, it should be considered immutable
```
Comment (JaredSnider-Bitwarden) [inline test/Core.Test/Auth/Services/AuthRequestServiceTests.cs:587]: "destiny" made me 🤣

File: src/Core/Auth/Services/Implementations/AuthRequestService.cs [hunk 2]
```diff
@@ -91,6 +129,40 @@ public async Task<AuthRequest> CreateAuthRequestAsync(AuthRequestCreateRequestMo
...
+            if (organizationUsers.Count == 0)
+            {
+                throw new BadRequestException("User does not belong to any organizations.");
+            }
+
+            AuthRequest? firstAuthRequest = null;
+            foreach (var organizationUser in organizationUsers)
+            {
+                await _eventService.LogOrganizationUserEventAsync(organizationUser, EventType.OrganizationUser_RequestedDeviceApproval);
+                var createdAuthRequest = await CreateAuthRequestAsync(model, user, organizationUser.OrganizationId);
```
[the remark this answers is not in the document]
Comment (JaredSnider-Bitwarden) [reply src/Core/Auth/Services/Implementations/AuthRequestService.cs:150]: Per product, no we are not. Might change plans in the future though.

**Review by JaredSnider-Bitwarden — COMMENTED** [review]
Great work! Just a few questions!

File: src/Core/Auth/Services/Implementations/AuthRequestService.cs [hunk 3]
```diff
@@ -52,13 +61,42 @@ public class AuthRequestService : IAuthRequestService
     public async Task<AuthRequest?> GetValidatedAuthRequestAsync(Guid id, string code)
```
[the remark this answers is not in the document]
Comment (JaredSnider-Bitwarden) [reply src/Core/Auth/Services/Implementations/AuthRequestService.cs:61]: These functions could also be leveraged elsewhere in the file. 
Note: if this refactor proves too time consuming b/c you have to touch so many tests and update things... then proceed with the working code you have!

[the remark this answers is not in the document]
Comment (justindbaur) [reply src/Core/Auth/Services/Implementations/AuthRequestService.cs:61]: I believe I implemented the spirit of most of this, I named the expiration checker a little more generic but was able to use it in a lot of places and I think it improves things a lot. Let me know if I implemented this stuff to your liking!

**Review by JaredSnider-Bitwarden — APPROVED** [review]
Super clean! Great work on this refactor!

**Review by JaredSnider-Bitwarden — APPROVED** [review]
LGTM!
